mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -668,8 +727,7 @@ def log_batch(self, run_id, metrics, params, tags):
         try:
             for param in params:
                 self.log_param(run_id, param)
-            for metric in metrics:
-                self.log_metric(run_id, metric)
+            self.log_metrics(run_id, metrics)
             for tag in tags:
                 self.set_tag(run_id, tag)
```

Comment:
this change only batches metrics -- params and tags are still added one at a time.

I am curious -- was there a particular reason you chose not to implement batch insert for params and tags. Would be easier than metrics.

Reply:
Yes definitely - this can be done in the same way for params and tags as well. I just started with metrics because it influences the runtime the most. Handling the constraints might be a little bit more tricky for params and tags.